### PR TITLE
fallback: replace use of `backoff` crate of by own implementation

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by ilyagr** [comment]
Are you planning to move from `chrono-english` to [`interim`](https://github.com/conradludgate/interim) in this PR (as the advisory suggests)? Did that turn out to be less trivial than the advisory makes it sound?

**Comment by martinvonz** [comment]
> Are you planning to move from `chrono-english` to [`interim`](https://github.com/conradludgate/interim) in this PR (as the advisory suggests)? Did that turn out to be less trivial than the advisory makes it sound?

Yes, I'm working on that now. It's not as trivial as it sounds.

File: lib/src/lock/fallback.rs [hunk 1]
```diff
@@ -17,58 +17,79 @@ use std::fs::OpenOptions;
...
+                            && err.kind() == std::io::ErrorKind::PermissionDenied) =>
+                {
+                    if let Some(duration) = backoff_iterator.next() {
+                        std::thread::sleep(duration);
+                    } else {
+                        panic!(
+                            "Timed out while trying to create lock file {}: {}",
+                            path.display(),
+                            err
+                        );
```
Comment (yuja) [inline lib/src/lock/fallback.rs:79]: nit: we should remove panic!() (by a separate patch)

(No action needed)

**Review by yuja — APPROVED** [review]
LG.

For `chrono-english`, I think it's also good to implement minimal set of date parsing which we really need.

**Comment by martinvonz** [comment]
I added a temporary ignore of the `chrono-english` crate to unblock CI and other PRs.

**Pull request merged**